feat(cli): use vite to load auth config when we detect a vite project

shreyassanthu77 opened the pull request:
When using frameworks that build on top of vite, some imports may not be resolved correctly when using jiti (like path aliases, virtual modules etc)

so this pr uses vite to try and resolve a config file and if one exists we use that with all the plugins enabled to load the auth config file so that will make sure all the imports are resolved properly and any vite specific logic need not be stubbed out just to use the better auth cli

## Notes
- splits out the config file path resolution logic from the actual loading of the file
- added vite as a dev dependency for type checking
- loads the actual vite module from the user's project to load the config
- falls back to jiti if vite is not found, package json is not found or if vite could not load a vite config 

## Few Considerations
- the change currently expects a few APIs from node and vite to be available for it to work
    1. we expect the runnerImport API to be available in vite but may not be available in older vite versions, needs better version resolution which currently is not implemented
    1. the runnerImport API used to load the config is marked experimental tho the api didn't change since 6.2
- some frameworks like astro and solid start wrap vite so we might need special handling for them (or leave it as is and just let it fallback to jiti?)
- no tests have been implemented for the new vite specific code

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Use Vite v6–v7 to load the auth config in Vite projects so aliases and plugin-based imports resolve correctly, with a safe fallback to c12/Jiti when Vite isn’t available or fails.

- **New Features**
  - Auto-detect Vite and load the auth config with the project’s Vite (plugins on).
  - Resolve TS path aliases, virtual modules, and framework imports.
  - Fallback to c12/Jiti if Vite is missing, runnerImport is unavailable, or Vite load fails.

- **Refactors**
  - Split config path resolution from loading; search paths relative to cwd.
  - Stronger validation and errors: try/catch around Vite load, export checks after both loaders, unified “server-only” guidance, and deduped messages.
  - Add Vite dev dep and optional peer; load user’s Vite at runtime via package.json resolution.
  - Tests cover Vite alias resolution and fallback to c12/Jiti when no Vite config is present.

<sup>Written for commit e8d02516309694674a3357db59f5b557eb20cbfc. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

shreyassanthu77 committed:
chore(refactor): split config file resolution and loading in the cli

shreyassanthu77 committed:
feat: use vite to resolve the auth config when available

himself65 reviewed (commented):
I think the idea is cool. Probably, we can remove Jiti? @Bekacru

shreyassanthu77 committed:
fix: use paths relative to cwd when searching for the auth config

himself65 commented:
> I think we can drop the special case svelte kit handling
> 
> been using basically this same code (well with an older better auth version but still) for about ~a year~ 6 months :) and seems to be working well

What's that?

shreyassanthu77 commented:
uhmm can you do it please 😅 I have a lot of stuff to do this week

or merge this and I'll make a new pr next week when I'm free that is also fine

himself65 commented on packages/cli/src/utils/get-config.ts:
```diff
@@ -162,6 +165,111 @@ const isDefaultExport = (
+async function loadConfigWithVite(cwd: string, resolvedConfigPath: string) {
+	const packageJsonPath = path.join(cwd, "package.json");
+	if (existsSync(packageJsonPath)) {
+		const require = createRequire(packageJsonPath);
```
What's the reason we don't use import('vite') here?

shreyassanthu77 commented:
> I just realized that this might affect https://github.com/better-auth/better-auth/pull/6120, https://github.com/better-auth/better-auth/pull/6113, since we use vitest for testing

why is that?

shreyassanthu77 committed:
chore(cli): fix lockfile

himself65 reviewed (commented):
I think we need to add vite to peer deps

shreyassanthu77 commented:
@himself65 regarding #6373 you want me to try the esbuild thing?

Also recently started an astro project and ran into this issue myself because my pr only works with a pure vite projects lol. I looked into it and astro wraps vite with their own cli without a real public api... which is kinda sad

shreyassanthu77 commented:
heyyy @himself65 sorry for the ping but is there anything i can do to get this pr merged? happy to fix anything or change things

shreyassanthu77 committed:
fix(cli): improve config loading robustness and validation

- Remove duplicate error message in resolveConfigFilePath
- Wrap vite config loading in try/catch to gracefully fall back to c12/jiti
- Move auth export validation after all loaders so it applies to both vite and c12 paths

shreyassanthu77 committed:
test(cli): add tests for vite config loading path

- Test that vite-specific resolve aliases work (proves vite is the loader, not c12/jiti)
- Test fallback to c12 when no vite config is present

shreyassanthu77 commented:
@Bekacru rebased and fixed all the issues and also added tests
